### PR TITLE
Turn articles section into an h-feed

### PR DESCRIPTION
**Pull request by Zegnat**
Per https://github.com/ScalaWilliam/ScalaWilliam.com/issues/64#issuecomment-318834486 this turns the articles section of the homepage into an h-feed.

It will currently parse the author as `https://www.scalawilliam.com/`, of course an h-card is necessary at that domain for parsers to do more fancy stuff with that. Discussions about [authorship detection for a feed](https://indieweb.org/authorship#Questions) are still ongoing, therefor I am not doing anything fancy and just explicitly saying who the author is.

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -42,35 +42,28 @@
...
-        <li><a href="/1609/feature-switches-agile-scala-jmx/">Feature Switches, Inheritance and Agile with Scala &amp; JMX on the JVM</a></li>
-    </ul>
-</section></div>
+    <div class="section h-feed" id="articles">
+        <section class="slide">
+            <h2 class="p-name"><data class="u-url" value="https://www.scalawilliam.com/#articles">Articles</data></h2>
+            <a class="u-author" href="/" style="display:none"></a><!-- Help parsers out. The author of all of these articles is https://www.scalawilliam.com/ -->
+            <p>Also on <a href="https://plus.google.com/collection/oFuHlB" target="_blank">Google+</a>.</p>
+            <ul>
+                <li class="h-entry"><a class="p-name" href="https://dev.to/scalawilliam/unit-testing-is-simple">Unit testing is simple</a>, <time class="dt-published" datetime="2017-05-04">May 4, 2017</time></li>
```
Comment (ScalaWilliam) [inline index.php]: might be easier to maintain with `<time>` tag being split off into a new line.

Also I don't need precise dates, "May 2017" would do well as a format used in the rest of the page.

Comment (Zegnat) [inline index.php]: Would you still want to provide the more precise dates internally on the `datetime` attribute or not?

I would recommend keeping the metadata as precise as possible. For archiving reasons as well as to make sorting easier (either just as a future site update or for microformats consumers).

File: index.php [hunk 2]
```diff
@@ -42,35 +42,28 @@
...
+                <li class="h-entry"><a class="p-name" href="https://dev.to/scalawilliam/2017-freelance-development-platforms">2017 freelance development platforms</a>, <time class="dt-published" datetime="2017-04-20">Apr 20, 2017</time></li>
```
Comment (ScalaWilliam) [inline index.php]: Should there be `u-uid` or `u-url` or something like that?

File: index.php (hunk 1 above)
Comment (ScalaWilliam) [inline index.php]: For the computer, yes. For the visual reader, no.

File: index.php (hunk 2 above)
Comment (Zegnat) [inline index.php]: > Should there be u-uid or u-url or something like that?

* `u-uid` is really only a good idea if you are moving things around a lot and URLs cannot be used as identifiers. Also, for unique IDs to work, you set them once for life. I wouldn’t bother with it.

* `u-url` is [implied](http://microformats.org/wiki/microformats2-parsing#parsing_for_implied_properties). If there is only 1 `a`-element as a direct child of the microformats object (the `h-entry` in this case), its `href` is taken as `url` property:
  
  > * else if .h-x>a[href]:only-of-type:not[.h-*], then use that [href] for url

Comment (ScalaWilliam) [inline index.php]: All right, understood, thanks.

**Comment by Zegnat** [comment]
Comments addressed. Also sorted the list based on date (that is: moving April to its correct place).

**Comment by ScalaWilliam** [comment]
> might be easier to maintain with `<time>` tag being split off into a new line.

Ran-away comment :-)

**Comment by Zegnat** [comment]
Rebased on latest index changes (link to `/1705/scala-android-opportunity/`) and put `<time>` on its own line.

**Pull request merged**